### ISSUE TITLE
getAllByIndex observable won't resolve if cursor has value

### ISSUE THREAD

**Issue by desmondchou** [assuncaocharles/ngx-indexed-db#209]
I have an objectStore 'activity' with index 'company_id'
Insert 2 entries with same 'company_id' say 'id1'

getAllByIndex('activity', 'company_id', IDBKeyRange.only(id1').subscribe((result) => console.log(result));

This observable will never be resolved. I'm wondering we are not listening to cursor onsuccess event for processing subsequent records / get notified when the cursor reaches the end.

`    getAllByIndex(storeName, indexName, keyRange) {
        const data = [];
        return from(new Promise((resolve, reject) => {
            this.openCursorByIndex(storeName, indexName, keyRange)
                .pipe(take(1))
                .subscribe((event) => {
                const cursor = event.target.result;
                if (cursor) {
                    data.push(cursor.value);
                    cursor.continue();
                }
                else {
                    resolve(data);
                }
            }, (reason) => reject(reason));
        }));
    }`

**Comment by assuncaocharles** [comment]
Can you provide a codesandbox with the problem?

**Comment by rinckd** [comment]
Love your project, and I just found it yesterday. I encountered this situation. 

If you have something like this:
```
  objectStoresMeta: [
    {
      store: 'people',
      storeConfig: { keyPath: 'id', autoIncrement: true },
      storeSchema: [
        { name: 'name', keypath: 'name', options: { unique: false } },
        { name: 'workerType', keypath: 'workerType', options: { unique: false } },
        { name: 'email', keypath: 'email', options: { unique: false } },
      ],
    },
```

and you add two people:
```
    this.dbService
      .add('people', {
        name: `charles number ${Math.random() * 10}`,
        email: `email number ${Math.random() * 10}`,
        workerType: `office`,
      })
      .subscribe((result) => {
        console.log('result: ', result);
      });
```

this will not work
```
    this.dbService.getAllByIndex('people', 'workerType', IDBKeyRange.only('office')).subscribe((val) => {
      console.log(val);
    });
```

I have confirmed that the PR fixes this issue.

**Comment by codafish-gmbh** [comment]
Same problem here. getAllByIndex observable won't be resolved even though data is available in the store.

**Comment by codafish-gmbh** [comment]
Hi there. I don't want to bother anyone. But is there any chance that fix #210 will be merged in the project? Thank you so much!

**Comment by assuncaocharles** [comment]
I wanna merge it as soon some test will be added for it. Feel free to added it  :)
